### DISCUSSION TITLE
Why is `Sentry.currentHub.options` internal?

### DISCUSSION THREAD

**Discussion by alestiago** [getsentry/sentry-dart#2073] [General]
`Sentry.currentHub.options` is marked as internal, however that same `SentryOptions` object can be retrieved during initialisation 

```dart
 await Sentry.init((options) {});
```

Is there any particular reason on why `Sentry.currentHub.options` is marked as internal? Could such reason be documented in the source code?

I've seen there was an initiative to remove `Sentry.currentHub`  (https://github.com/getsentry/sentry-dart/issues/405), I'm wondering what happened with such effort (since v6 https://github.com/getsentry/sentry-dart/discussions/419 was made in 2021) and if there are any plans for the future?

**Comment by buenaflor** [comment]
Hey, do you have any specific use case in mind for retrieving the options besides in the init?

Tracing back the change for `Sentry.currentHub` it was re-added when they were working on the performance API, I joined recently so tbh I'm not sure what the context back then was for removing it.

**Reply by ueman** [reply]
`currentHub` is not really considered to be part of the public API to make changes easier, I guess. The options should not be adjusted anywhere else than during init, so that's why they're not exposed. As the comment on https://github.com/getsentry/sentry-dart/issues/405 points out, you should just use the static Sentry class instead, since it provides all the same functionality. I don't recall that there's more to it.

**Reply by buenaflor** [reply]
Yeh makes sense, was just a bit confused since in the v6 changelog it was stated as removed under breaking change

**Reply by ueman** [reply]
It was indeed removed, only to be added again later. I believe it was to enable the other Sentry libraries (files, dio, etc) to read from the options since they can't be initialized during the Sentry init call. 

dio for example reads whether breadcrumbs should be added from the options. Without access to the options it can't decide that, so it would need to be configured while setting up dio with Sentry, which is somewhat against the SDK philosophy. 

However, integrations from other people usually don't need that since their config can't be added to the options, and things like performance tracing properly bails out if it's disabled, so there's no need to read that from the options, since you can just use the static API.

**Comment by alestiago** [comment]
Closing this discussion since answers were given.
